main.go

```diff
@@ -82,6 +108,155 @@ func runMain() error {
 		return nil
 	}
+	
+	// Guess the next release version
+	latestReleaseVersion := getLatestReleaseVersion(repository, NO_PREVIOUS_VERSION)
+
+	// Conduct changelog file validation
+	changelogFilepath := path.Join(currentWorkingDirectory, relChangelogFilepath)
+
+	tbdHeaderCount := grepFile(changelogFilepath, TBD_VERSION_HEADER_REGEX)
+	if tbdHeaderCount != EXPECTED_NUM_TBD_HEADER_LINES {
+		fmt.Printf("There should be %d TBD header lines in the changelog. Instead there are %d.\n", EXPECTED_NUM_TBD_HEADER_LINES, tbdHeaderCount)
+		return nil
+	}
+
+	versionHeaderCount := grepFile(changelogFilepath, VERSION_HEADER_REGEX)
+	if versionHeaderCount == 0 {
+		fmt.Println("No previous changelog versions were detected in this changelog. Are you sure that the changelog is in sync with the release tags on this branch?")
+		return nil
+	}
+
+	existsBreakingChanges := detectBreakingChanges(changelogFilepath)
+	var nextReleaseVersion semver.Version
+	if existsBreakingChanges {
+		nextReleaseVersion = latestReleaseVersion.IncMinor()
+	} else {
+		nextReleaseVersion = latestReleaseVersion.IncPatch()
+	}
 
+	go wait()
+	fmt.Printf("VERIFICATION: Release new version '%s'? (ENTER to continue, Ctrl-C to quit))", nextReleaseVersion.String())
+	fmt.Scanln()
 	return nil
 }
+
+func wait() {
+    i := 0
+    for {
+        time.Sleep(time.Second * 1)
+        i++
+    }
+}
+
+// adapted from: https://stackoverflow.com/questions/26709971/could-this-be-more-efficient-in-go
+func grepFile(file string, regexPat string) int64 {
+	r, err := regexp.Compile(regexPat)
+	if err != nil {
+		logrus.Errorf("Could not parse regexp: '%s'", regexPat, err)
+	}
+    patCount := int64(0)
+    f, err := os.Open(file)
+    if err != nil {
+		logrus.Errorf("An error occurred while attempting to open file", err)
+    }
+    defer f.Close()
+    scanner := bufio.NewScanner(f)
+    for scanner.Scan() {
+		if r.Match(scanner.Bytes()) {
+			patCount++
+		}
+    }
+    if err := scanner.Err(); err != nil {
+        logrus.Errorf("An error occurred while scanning file.\n%v", err)
+    }
+    return patCount
+}
+
+func getLatestReleaseVersion(repo *git.Repository, noPrevVersion string) semver.Version {
+	semverRegex, err := regexp.Compile(SEMVER_REGEX)
+	if err != nil {
+		logrus.Errorf("Could not parse regexp: '%s'", SEMVER_REGEX, err)
+	}
+
+	tagrefs, err := repo.Tags()
+	if err != nil {
+		logrus.Errorf("An error occurred while retrieving tags for repository", err)
+	}
+
+	// Trim tagrefs and filter for only tags with X.Y.Z version format
+	var tagSemVers []*semver.Version
+	tagrefs.ForEach(func(tagref *plumbing.Reference) error {
+		tagName := tagref.Name().String()
+		tagName = strings.ReplaceAll(tagName, tagsPrefix, "")
+
+		if semverRegex.Match([]byte(tagName)) {
+			tagSemVer, err := semver.StrictNewVersion(tagName)
+			if err != nil {
+				logrus.Errorf("An error occurred while converting tags to semantic version.", err)
```

Comment:
1) You'll want to return a `stacktrace.Propagate` here
2) You'll definitely want to include the tag that failed in the error message, else when this condition does get tripped eventually you'll get a message that's basically like "An  error occurred" and you'll be left scratching your head as to which tag failed (this is https://github.com/kurtosis-tech/kurtosis-internal-docs/blob/master/general-style-guide.md#when-propagating-errors-log-the-parameters-of-the-function-that-failed in the style guide)